src/libos/src/time/mod.rs

```diff
@@ -149,21 +149,55 @@ pub fn do_clock_getres(clockid: ClockId) -> Result<timespec_t> {
     Ok(res)
 }
 
+const TIMER_ABSTIME: i32 = 0x01;
+
 pub async fn do_nanosleep(req: &timespec_t, rem: Option<&mut timespec_t>) -> Result<()> {
+    do_clock_nanosleep(ClockId::CLOCK_REALTIME, 0, req, rem).await?;
+    Ok(())
```

Comment:
Maybe just return the function result here.

Reply:
Yeap.